### ISSUE TITLE
[Linux][mp] Crash when encountering certain NPCs

### ISSUE THREAD

**Issue by 4LT** [ValveSoftware/source-sdk-2013#443]
Hello, I'm encountering a crash in our mod when encountering certain NPCs, apparently npc_citizen but there may be others.  One replicable point where this happens is on d3_c17_13, when the player encounters the 2nd and 3rd rebels, who are hiding adjacent to the first parking garage.  This can be re-created by copying the map from HL2 to Source SDK Base 2013 Multiplayer, so it's not mod-specific.  A backtrace leads to studiorender.so.  This crash does not occur in Windows.

**Comment by kisak-valve** [comment]
Hello @4LT, please copy your system information from steam (```Steam``` -> ```Help``` -> ```System Information```) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report. In particular, if you can find any CrashIDs or minidumps related to this issue, please include them.

**Comment by 4LT** [comment]
I posted a gist. Can't find any minidumps (```find . | grep mdmp``` revealed nothing)

**Comment by kisak-valve** [comment]
On linux, minidumps from steam are commonly found in `/tmp/dumps/` and end with .dmp.

**Comment by 4LT** [comment]
Thanks, uploaded dump file.

**Comment by kisak-valve** [comment]
For reference, the attached minidump is indeed a SIGSEGV in studiorender.so.

**Comment by Yetoo1** [comment]
@kisak-valve This seems like a duplicate of #2225.

**Comment by TravisWehrman** [comment]
Some quick debugging found the segfault occurs in [src/public/materialsystem/imesh.h#L1417](https://github.com/ValveSoftware/source-sdk-2013/blob/0d8dceea4310fde5706b3ce1c70609d72a38efdf/mp/src/public/materialsystem/imesh.h#L1417).

Top-level function is `DrawModel()`, the  models I see being processed here are the `Humans/Group03m/*.mdl` variants.

Unsure if it's an incorrect SIMD asm implementation, but modifying `studiorender.so::r_studiodraw.cpp::R_StudioSoftwareProcessMesh()` to use `FastVertex()` instead of `FastVertexSSE()` in the two instances appears to effectively bypass this segfault. We'll be using this workaround for our mod. Presumed perf hit due to using the non SSE version `FastVertex()` appears unmeasurable thus inconsequential for our needs.

I used the aforementioned `d3_c17_13` and the end of `d1_canals_01` maps as test cases for this issue.

**Comment by CTX808** [comment]
> Some quick debugging found the segfault occurs in [src/public/materialsystem/imesh.h#L1417](https://github.com/ValveSoftware/source-sdk-2013/blob/0d8dceea4310fde5706b3ce1c70609d72a38efdf/mp/src/public/materialsystem/imesh.h#L1417).
> 
> Top-level function is `DrawModel()`, the models I see being processed here are the `Humans/Group03m/*.mdl` variants.
> 
> Unsure if it's an incorrect SIMD asm implementation, but modifying `studiorender.so::r_studiodraw.cpp::R_StudioSoftwareProcessMesh()` to use `FastVertex()` instead of `FastVertexSSE()` in the two instances appears to effectively bypass this segfault. We'll be using this workaround for our mod. Presumed perf hit due to using the non SSE version `FastVertex()` appears unmeasurable thus inconsequential for our needs.
> 
> I used the aforementioned `d3_c17_13` and the end of `d1_canals_01` maps as test cases for this issue.

Yeah the inline asm looks off. Looks like it copies 64 bytes in the linux version compared to the win32 version and the structure is only 48 bytes (compile time assert for the sizeof the struct wouldn't hurt). Anyways here's a version with intrinsics which you can use for both msvc and gcc:

`	const unsigned char *pSrc = ( unsigned char* )&vertex;
	const __m128 xmm0 = _mm_load_ps( ( float* )pSrc );
	const __m128 xmm1 = _mm_load_ps( ( float* )( pSrc + 16 ) );
	const __m128 xmm2 = _mm_load_ps( ( float* )( pSrc + 32 ) );
	const unsigned char *pDst = ( unsigned char* )m_pCurrPosition;
	_mm_stream_ps( ( float* )pDst, xmm0 );
	_mm_stream_ps( ( float* )( pDst + 16 ), xmm1 );
	_mm_stream_ps( ( float* )( pDst + 32 ), xmm2 );`
